### ISSUE TITLE
Some BENTOML prometheus /metrics appear duplicated

### ISSUE THREAD

**Issue by marcindulak** [bentoml/BentoML#2286]
**Describe the bug**

It seems like certain metrics are duplicated.

**To Reproduce**

Steps to reproduce the issue:
1. start the service
```sh
git clone https://github.com/bentoml/gallery
cd gallery/quickstart/
git checkout c03e7604581265fc81247556a455cce11d03f686
python3 -m venv venv
source venv/bin/activate
pip install bentoml==1.0.0a4 scikit-learn==1.0.2
python train.py
bentoml serve iris_classifier:svc --reload
```
2. in another terminal issue a request
```sh
curl -X POST -H "content-type: application/json" --data "[5, 4, 3, 2]" http://127.0.0.1:5000/classify
```
3. check the /metrics
```sh
curl -s http://localhost:5000/metrics | grep request_total | grep classify | grep 'http_response_code="2'
BENTOML_iris_classifier_request_total{endpoint="/classify",http_response_code="200",service_version=""} 1.0
BENTOML_iris_classifier_request_total{endpoint="/classify",http_response_code="200",service_version=""} 1.0
```

**Expected behavior**

Single metrics (?).

**Screenshots/Logs**

**Environment:**
 - OS:
 ```sh
cat /etc/*release | grep PRETTY
PRETTY_NAME="Ubuntu 20.04.3 LTS"
```
 - Python Version
 ```sh
python --version
Python 3.8.10
```
 - BentoML Version https://github.com/bentoml/BentoML/tree/v1.0.0-a4

**Additional context**

Several metrics appear duplicated (last 3 line pairs)
```sh
curl -s http://localhost:5000/metrics | grep TYPE | sort
# TYPE BENTOML_iris_classifier_request_created gauge
# TYPE BENTOML_iris_classifier_request_duration_seconds_created gauge
# TYPE BENTOML_iris_classifier_request_duration_seconds histogram
# TYPE BENTOML_iris_classifier_request_duration_seconds histogram
# TYPE BENTOML_iris_classifier_request_in_progress gauge
# TYPE BENTOML_iris_classifier_request_in_progress gauge
# TYPE BENTOML_iris_classifier_request_total counter
# TYPE BENTOML_iris_classifier_request_total counter
```

**Comment by timliubentoml** [comment]
Thanks! I've verified that this is an issue and that it is reproducible. We've scheduled some time to work on it

**Comment by parano** [comment]
This has been fixed in https://github.com/bentoml/BentoML/pull/2394, will be available in the upcoming release 1.0.0a7
